### ISSUE TITLE
Mulitple outputs with weighted loss?

### ISSUE THREAD

**Issue by fufrank5** [tflearn/tflearn#484]
The loss of the objective function would be like loss = loss_output1 + weight2 * loss_output2.
Is there any way to implement this under the current framework?

Thanks,
Lisheng

**Comment by aymericdamien** [comment]
Yes, DNN class accept a function as argument. It must take 2 arguments: "logits" and "targets" as input and return the loss value.

**Comment by fufrank5** [comment]
I understand that the loss function can be user-defined and passed to the regression layer. However, I haven't found a way to pass multiple targets (outputs) to the regression layer. Is there any example like this? Thanks.

**Comment by henryem** [comment]
It depends on exactly what you mean by multiple outputs.

If you only have a small fixed number (say, 2) then create each loss layer separately (siblings in the graph).  Then create a layer that is a fixed linear combination of those loss layers, using a fixed array of weights to parameterize the linear combination.  (It will be a parent of the individual losses.)  Pass that loss to the regression layer.

If you have lots of outputs (say K, so your output is an NxK matrix), then you could do something like this (in this example I'm computing the L2-squared loss) (note that I haven't tested this exact code, so there might be bugs):

    errorWeightsArray = [weight1., weight2, <...>] # Fill in your weights here.
    errorWeights = tf.constant(errorWeightsArray, shape=[1, len(errorWeightsArray)])
    squaredErrors = tf.pow(outputLayer - trueOutputsPlaceholder, 2)
    mseByOutput = tf.reduce_mean(squared_errors, 0) # A 1xK tensor
    weightedMse = tf.reduce_sum(tf.mul(mseByOutput, weightTensor))
    # ...use weightedMse as your loss tensor.
